scripts/benchmark_ketjl.jl

```diff
@@ -149,3 +149,423 @@ for (sys, id) in zip(sys_list, ids)
+    dim_perm_group[key] = @benchmarkable begin
+        mat = randn(ComplexF64, $size, $size)
+        result = permute_systems(mat, $perm_jl, $dims)
+    end
+end
+    
+vec_group = SUITE["TestPermuteSystemBenchmarks"]["test_bench__permute_systems__vary__vector_input"]
+
+vec_cases = [
+    (4, [2, 2], [1, 0]),
+    (8, [2, 2, 2], [1, 2, 0]),
+    (16, [4, 4], [1, 0]),
+    (64, [4, 4, 4], [2, 1, 0]),
+    (256, [16, 16], [1, 0]),
+]
+
+for (size, dims, perm) in vec_cases
+    perm_jl = [ p+1 for p in perm]
+    key = "test_bench__permute_systems__vary__vector_input[$size-$dims-$perm]"
+    vec_group[key] = @benchmarkable begin
+        vec = randn(ComplexF64, $size)
+        result = permute_systems(vec, $perm_jl, $dims)
+        @assert length(result) == $size
+    end
+end
+
+# ---- choi---- #
+
+SUITE["TestKraustoChoiBenchmarks"] = BenchmarkGroup()
+SUITE["TestKraustoChoiBenchmarks"]["test_bench__choi_representation__vary__kraus_ops"] = BenchmarkGroup()
+SUITE["TestKraustoChoiBenchmarks"]["test_bench__choi_representation__param__sys"] = BenchmarkGroup()
+
+choi_group = SUITE["TestKraustoChoiBenchmarks"]["test_bench__choi_representation__vary__kraus_ops"]
+
+# (dim, num_ops, cp)
+cases = [
+    (2, 4, true),
+    (2, 32, true),
+    (4, 4, true),
+    (4, 32, true),
+    (16, 4, true),
+    (16, 32, true),
+    # non-CP cases are NOT supported in ketjl.
+]
+
+for (dim, num_ops, cp) in cases
+    key = "test_bench__choi_representation__vary__kraus_ops[$dim-$num_ops-True]"
+    if cp
+        choi_group[key] = @benchmarkable begin
+            K = [randn(ComplexF64, $dim, $dim) for _ in 1:$num_ops]
+            C = choi(K)
+            @assert size(C) == ($dim^2, $dim^2)
+        end
+    end
+end
+
+sys_group = SUITE["TestKraustoChoiBenchmarks"]["test_bench__choi_representation__param__sys"]
+
+# Only sys==2 is supported,
+key = "test_bench__choi_representation__param__sys[2]"
+sys_group[key] = @benchmarkable begin
+    dim = 16
+    num_ops = 4
+    K = [randn(ComplexF64, dim, dim) for _ in 1:num_ops]
+    C = choi(K)
+    @assert size(C) == (dim^2, dim^2)
+end
+
+# ---- pauli ---- #
+SUITE["TestPauliBenchmarks"] = BenchmarkGroup()
+SUITE["TestPauliBenchmarks"]["test_bench__pauli__vary__ind"] = BenchmarkGroup()
+
+pauli_group = SUITE["TestPauliBenchmarks"]["test_bench__pauli__vary__ind"]
+
+for j in 0:3
+    key1 = "test_bench__pauli__vary__ind[int-$j]"
+    pauli_group[key1] = @benchmarkable begin
+        M = pauli($j)
+        @assert size(M) == (2,2)
+    end
+end
+
+for n in (2, 4, 8)
+    key2 = "test_bench__pauli__vary__ind[list-$n]"
+    inds = rand(0:3, n)
+    pauli_group[key2] = @benchmarkable begin
+        M = pauli($inds)
+        @assert size(M) == (2^$n, 2^$n)
+    end
+end
+
+# ---- permutation_matrix---- #
+
+SUITE["TestPermutationOperatorBenchmarks"] = BenchmarkGroup()
+
+dim_perm_group = SUITE["TestPermutationOperatorBenchmarks"]["test_bench__permutation_operator__vary__dim_perm"] = BenchmarkGroup()
+
+cases = [

```

Comment:
As you have done elsewhere, could you have a comment that states what these arguments correspond to?